src/app/register/supplier/FormComponent.tsx

```diff
@@ -30,6 +39,22 @@ const FormComponent = () => {
     },
   })
 
+  const getSupplierCatogires = async () => {
+    const res:any = await Sup.getSupplierCategories()
+    return res
+  };
+  
+  useEffect(()=> {
+    getSupplierCatogires()
+    .then((res) => {
+      setCategories(res.data.supplierCategories.results),
+      setFirstCategory(res.data.supplierCategories.results[0].id)
+    })
+    .catch((error)=> console.log(error))
+    setInitialValues( prev => ({...prev, supplierCategory: firstCategory}))
+  }
+  ,[firstCategory]);
```

Comment:
this property from useEffect don't make a infinite lopping? because u set first category in all fetch successful. 
Resolve this pls.

Reply:
i tested with console log and they doesn`t make de infint loop. but they also dont need this to work... thats my mistake.